packages/terra-infinite-list/src/InfiniteList.jsx

```diff
@@ -0,0 +1,435 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import ResizeObserver from 'resize-observer-polyfill';
+import List from 'terra-list';
+import SelectableUtils from 'terra-list/lib/SelectableUtils';
+import InfiniteUtils from './_InfiniteUtils';
+import styles from './InfiniteList.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The child list items, of type InfiniteList.Item, to be placed within the infinite list.
+   * For further documentation of InfiniteList.Item see terra-list's ListItem.
+   */
+  children: PropTypes.node,
+  /**
+   * Whether or not unselected items should be disabled.
+   * Helpful for enabling max row selection.
+   */
+  disableUnselectedItems: PropTypes.bool,
+  /**
+   * Whether or not the child list items has a disclosure indicator presented.
+   * The behavior is intended to be used with a single selection style list, as multi selection style list should not perform disclosures.
+   */
+  hasChevrons: PropTypes.bool,
+  /**
+   * An indicator to be displayed when no children are yet present.
+   */
+  initialLoadingIndicator: PropTypes.element,
+  /**
+   * Whether or not the child list items should have a border color applied.
+   */
+  isDivided: PropTypes.bool,
+  /**
+   * Determines whether or not the loading indicator is visible and if callbacks are triggered.
```

Comment:
>  and if callbacks are triggered. 

Should we add a finished loading callback prop?

Reply:
The consumer handles loading, the callback we trigger is 'onRequestItems'.